R/functions_netCDF.R

```diff
@@ -456,42 +687,377 @@ create_netCDF_from_array_with_variables <- function(x, locations, grid,
+            
+            vals <-  data[, n] # by time chunk or var chunk
+
+            var_start <-  c(1, n)
+            
+            try(ncdf4::ncvar_put(nc, varid = var_names[k],
+                                 vals = vals,
+                                 start = var_start,
+                                 count = c(var_chunksizes, 1)))
+          }
+        }
+      }
+    }
+    
+    #  GRIDDED --------------------------------------------------------------
+    if(isGridded == TRUE) {
+      
+      # more checks based on dim of var ----------------------------------------
+      
+      if(has_T_timeAxis & !has_Z_verticalAxis) stopifnot(nc_var_dims[3] == nn)
+      if(!has_T_timeAxis & has_Z_verticalAxis) stopifnot(nc_var_dims[3] == nn)
+      
+      var_chunksizes <- c(nc_var_dims[1], nc_var_dims[2]) # lon, lat
+      
+      #  -----------------------------------------------------------------------
+      # add variable values ! --------------------------------------------------
+      # ------------------------------------------------------------------------
+      if(nvars > 1) {
+        temp <- grid_template
+        temp[val_grid_ids] <- data[, k]
+        vals <- matrix(temp,  ncol = var_chunksizes[2])
+        
+        var_start <-  c(1, 1)
+        
+        try(ncdf4::ncvar_put(nc, varid = var_names[k],
+                             vals = vals,
+                             start = var_start,
+                             count = var_chunksizes))
+        
+      } else {
+        if(has_T_timeAxis && has_Z_verticalAxis) {
+          
+          temp <- grid_template
+          
+          for(z in seq(z_chunksize)) { # by Z axis
+           if(verbose) message('Adding depth layer ', z)
+            for (t in seq_len(nn)) { # col by col - always time in this case
+              
+              temp[val_grid_ids] <- data[, t, z]
+              vals <- matrix(temp, ncol = as.numeric(var_chunksizes[2]))
+              
+              try(ncdf4::ncvar_put(nc, varid = var_names[k],
+                                   vals = vals,
+                                   start = c(1, 1, t, z), #x-y-t-z
+                                   count = c(var_chunksizes, 1, 1)))
+            }
+          }
+        } else {
+          # write values, col by col -- n values can rep dif. time, or depth
+          for (n in seq_len(nn)) {
+            
+            temp <- grid_template
+            temp[val_grid_ids] <- data[, n]
+            vals <- matrix(temp,  ncol = var_chunksizes[2])
+
+            var_start <-  c(1, 1, n)
+
+            try(ncdf4::ncvar_put(nc, varid = var_names[k],
+                                 vals = vals,
+                                 start = var_start,
+                                 count = c(var_chunksizes, 1)))
+          }
+        }
+      }
+    }
+    
     # Flush this step to the file so we dont lose it
     # if there is a crash or other problem
     ncdf4::nc_sync(nc)
```

Comment:
line 1063: nc_sync is no longer inside a loop and now defeats purpose --> move it inside the loops or remove

Reply:
removed